mock-mongodb-&-flights-mock-tests

nehaabrol87 opened the pull request:
@jonathanbarton @morrissinger  Eyes please.

Tomorrow and in a separate PR I will move the connection in the before each to utils.
I will figure out a way for the local db instance to use the data from mongo.js file for envs other than prod(Let us talk about this tomorrow) 

Oops do not merge need to fix few eslint issues!!!

nehaabrol87 committed:
mock-mongodb-&-flights-mock-tests

morrissinger commented on .gitignore:
```diff
@@ -6,6 +6,7 @@
 *.log
 dist
 .idea
+mongo.js
```
What is this? (Because it's in the gitignore, it's not in GitHub and I don't have the luxury to examine it myself.

nehaabrol87 replied:
It is the js file that is generated after the tests insert.update and remove data from the collections. It can be used to populate a db  with data, if needed.

nehaabrol87 replied:
Yes I can change that and we don't want to upload it to git right? I mean it will be created when we run tests automatically. Also can we have the tests run as part of docker compose up.
@morrissinger  Also will the docker-compose up watch for code changes as of course we would not want to build every time we make a change locally while we are building stuff.

(Apologies in advance if that came as dumb question from this docker noob)

morrissinger commented on server/tests/flight.test.js:
```diff
@@ -1,13 +1,106 @@
+      });
+    });
+
+    it('should deny unique constraint violations on insert', function(done) {
```
If you are concerned about being able to understand test failure, one thing you could do is break the several assertions out into separate tests, put common code in a `beforeEach()` block, and lump the entire thing in its own `describe()` block.

nehaabrol87 replied:
Nice idea , if both of us are thinking on same lines then maybe a folder insides tests named flights where indexes.js would have the assertions for indexes etc etc .Just checking if you meant something like this or did you mean something else.

morrissinger replied:
Interesting idea. That's a good idea as well. My thinking was that you have assertions on lines 56, 57, 65, 66, 76, and 77 which might benefit from being separated into three different tests (I imagine you could put 56 and 57 together, 65 and 66 togethr, and 76 and 77 together). By separating these six assertions across three tests, it might be easier to diagnose a failure in the future. If this creates a lot of additional code, you could lump these three tests inside a `describe()` block and add a `beforeEach()` block to keep setup routines that are needed in common across all three tests.

nehaabrol87 replied:
aah that makes sense.

nehaabrol87 commented:
@morrissinger  @jonathanbarton  
So I made a couple of changes

1)I created a new folder called mongoMock inside tests which individual tests can use(module is connect) to make a connection to the fake db
2)I renamed the mongoMock persistent file to data.js and placed it inside the mongo mock folder. It will not be loaded to git.
3)I extracted individual assertions to separate tests to make debugging a little easier.
4)I added the 'lint' task  to the gulp test  run sequence so that it is enforced when we run tests.

morrissinger commented:
The changes address my points in code review. Looks good to me!
